typescript/sdk/src/providers/transactions/submitter/ethersV5/types.ts

```diff
@@ -39,3 +40,43 @@ export const EV5ImpersonatedAccountTxSubmitterPropsSchema =
 export type EV5ImpersonatedAccountTxSubmitterProps = z.infer<
   typeof EV5ImpersonatedAccountTxSubmitterPropsSchema
 >;
+export type EvmIcaTxSubmitterProps = z.infer<
+  typeof EvmIcaTxSubmitterPropsSchema
+>;
+
+export const EvmIcaTxSubmitterInternalSubmitterConfigSchema = z
+  .discriminatedUnion('type', [
+    z.object({
+      type: z.literal(TxSubmitterType.JSON_RPC),
+    }),
+    z
+      .object({
+        type: z.literal(TxSubmitterType.GNOSIS_TX_BUILDER),
+      })
+      .merge(EV5GnosisSafeTxBuilderPropsSchema.omit({ chain: true })),
+    z
+      .object({
+        type: z.literal(TxSubmitterType.GNOSIS_SAFE),
+      })
+      .merge(EV5GnosisSafeTxSubmitterPropsSchema.omit({ chain: true })),
+    z
+      .object({
+        type: z.literal(TxSubmitterType.IMPERSONATED_ACCOUNT),
+      })
+      .merge(
+        EV5ImpersonatedAccountTxSubmitterPropsSchema.omit({ chain: true }),
+      ),
+  ])
+  .default({
+    type: TxSubmitterType.JSON_RPC,
+  });
+
+export const EvmIcaTxSubmitterPropsSchema = z.object({
+  chain: ZChainName,
+  owner: ZHash.optional(),
+  destinationChain: ZChainName,
```

Comment:
I am actually a bit confused by the relationship between the chains/owner and internal submitter
if the internal submitter is a safe submitter then the owner address would have to match? should we remove the chain and owner from this schema entirely?

Reply:
The chain field specifies which chain the ICA transactions will be sent from, while the owner address is the owner of the ICA so yes in case the internal submitter is a safe, the addresses should match. I think I can enforce this requirement in the zod validator.